homeassistant/components/switchbee/light.py

```diff
@@ -0,0 +1,148 @@
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SwitchBeeCoordinator
+from .entity import SwitchBeeDeviceEntity
+
+MAX_BRIGHTNESS = 255
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _hass_brightness_to_switchbee(value: int) -> int:
+    """Convert hass brightness to SwitchBee."""
+    sb_brightness = int(100 * value / MAX_BRIGHTNESS)
+    # SwitchBee maximum brightness is 99
+    return sb_brightness if sb_brightness != 100 else 99
+
+
+def _switchbee_brightness_to_hass(value: int) -> int:
+    """Convert SwitchBee brightness to hass."""
+    if value == 99:
+        value = 100
+    return round(value * MAX_BRIGHTNESS / 100)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up SwitchBee light."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        SwitchBeeLightEntity(switchbee_device, coordinator)
+        for switchbee_device in coordinator.data.values()
+        if switchbee_device.type == DeviceType.Dimmer
+    )
+
+
+class SwitchBeeLightEntity(SwitchBeeDeviceEntity[SwitchBeeDimmer], LightEntity):
+    """Representation of a SwitchBee light."""
+
+    _attr_color_mode = ColorMode.BRIGHTNESS
+    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+
+    def __init__(
+        self,
+        device: SwitchBeeDimmer,
+        coordinator: SwitchBeeCoordinator,
+    ) -> None:
+        """Initialize the SwitchBee light."""
+        super().__init__(device, coordinator)
+        self._attr_is_on = False
+        self._attr_brightness = 0
+
+        self._update_attrs_from_coordinator()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_attrs_from_coordinator()
+        super()._handle_coordinator_update()
+
+    def _update_attrs_from_coordinator(self) -> None:
+
+        coordinator_device = cast(
+            SwitchBeeDimmer, self.coordinator.data[self._device.id]
+        )
+        brightness = coordinator_device.brightness
+
+        # module is offline
+        if brightness == -1:
+            # This specific call will refresh the state of the device in the CU
+            self.hass.async_create_task(self.async_refresh_state())
+
+            # if the device was online (now offline), log message and mark it as Unavailable
+            if self._is_online:
+                _LOGGER.warning(
+                    "%s light is not responding, check the status in the SwitchBee mobile app",
+                    self.name,
+                )
+                self._is_online = False
```

Comment:
In a follow-up PR, I think this bit could be moved to `entity.py`, as a standalone method `self._is_now_offline()`